internal/tools/aws/s3.go

```diff
@@ -0,0 +1,167 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+
+	apps "github.com/mattermost/mattermost-plugin-apps/aws"
+	log "github.com/sirupsen/logrus"
+)
+
+// DownloadS3Object downloads the specified object from the specified S3 bucket.
+func DownloadS3Object(bucketName, object, dir string, session *session.Session) error {
+	file, err := os.Create(fmt.Sprintf("%s/%s", dir, object))
+	if err != nil {
+		return err
+	}
+
+	defer file.Close()
+
+	downloader := s3manager.NewDownloader(session)
+
+	_, err = downloader.Download(file,
+		&s3.GetObjectInput{
+			Bucket: aws.String(bucketName),
+			Key:    aws.String(object),
+		})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UploadStaticFiles is used to upload static files to the static S3 bucket.
+func UploadStaticFiles(staticFiles map[string]apps.AssetData, bundleName string, logger log.FieldLogger) error {
+	for staticFile, staticKey := range staticFiles {
+		bundleDir := fmt.Sprintf("%s/%s", os.Getenv("TempDir"), bundleName)
+		fileDir := fmt.Sprintf("%s/static/%s", bundleDir, staticFile)
```

Comment:
Golang has a library for joining paths like this that is a little less brittle if you want to use it instead: https://play.golang.org/p/xaJ09phuASe

Reply:
Great stuff, thanks a lot. I replaced it everywhere in the code :)